blocks/library/list/index.js

```diff
@@ -17,60 +17,45 @@ import { registerBlockType, source, createBlock } from '../../api';
 import Editable from '../../editable';
 import BlockControls from '../../block-controls';
 
-const { children, prop } = source;
+const { html, prop } = source;
 
 const fromBrDelimitedContent = ( content ) => {
 	if ( undefined === content ) {
 		// converting an empty block to a list block
 		return content;
 	}
-	const listItems = [];
-	listItems.push( createElement( 'li', [], [] ) );
-	content.forEach( function( element, elementIndex, elements ) {
-		// "split" the incoming content on 'br' elements
-		if ( 'br' === element.type && elementIndex < elements.length - 1 ) {
-			// if is br and there are more elements to come, push a new list item
-			listItems.push( createElement( 'li', [], [] ) );
-		} else {
-			listItems[ listItems.length - 1 ].props.children.push( element );
+
+	return content.split( /<br\s*?\/?>/ ).reduce( ( memo, item ) => {
```

Comment:
This will be one of the less nice things about this change. Now be can't look inside the content that easily... I don't feel comfortable with a regex to parse this, it will lead to bugs. What if there's nested  `<br>`s? Maybe we should convert this to DOM (like we do with paste) and split on them there.

Reply:
we can, I was also thinking of removing this transform entirely, it doesn't make sense to me. It made sense a while back because we were adding `br`'s by default when hitting enter, it's not the case anymore.

Reply:
I think it still makes sense as we still do allow content like that. I believe it was also a request with the old editor. It's also nice that it stays together when you convert it to text.

Reply:
@youknowriad I'll adjust it